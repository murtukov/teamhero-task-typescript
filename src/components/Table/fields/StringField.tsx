import React, {CSSProperties} from 'react';
import {IFieldProps} from "./types";
import {styles as sharedStyles} from "./styles";
import {createUseStyles} from "react-jss";

export interface IStringFieldProps extends IFieldProps {
    data?: string,
    bold?: boolean
}

function StringField({data, bold = false}: IStringFieldProps) {
    const classes = useStyles();
    const styles: CSSProperties = {};

    if (bold) {
        styles.fontWeight = 'bold';
    }

    return (
        <td className={classes.root} style={styles}>
            {data}
        </td>
    );
}

const useStyles = createUseStyles({
    ...sharedStyles
});

export default StringField;