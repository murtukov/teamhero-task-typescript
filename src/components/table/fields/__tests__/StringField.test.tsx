import React from "react";
import { render } from "@testing-library/react";
import StringField from "../StringField";

const text = 'This is a test string';

describe('StringField component', () => {
    it('matches snapshot', () => {
        const { container } = render(<StringField source='dummy' data={text} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders correct fontWeight style', () => {
        const { container } = render(
            <StringField source='dummy' bold />
        );

        expect(container.firstElementChild).toHaveStyle({fontWeight: 'bold'});
    });
})

