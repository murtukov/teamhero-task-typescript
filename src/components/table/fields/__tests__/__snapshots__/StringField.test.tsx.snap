// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringField component renders correctly 1`] = `
<div
  class="cell-0-2-1 cell-d0-0-2-2"
>
  This is a test string
</div>
`;
